core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -802,8 +802,8 @@ class KafkaApis(val requestChannel: RequestChannel,
       replicaManager.fetchMessages(
         fetchRequest.maxWait.toLong,
         fetchRequest.replicaId,
-        fetchRequest.minBytes,
-        fetchRequest.maxBytes,
+        Math.min(fetchRequest.maxBytes, fetchRequest.minBytes),
```

Comment:
I don't think thats right - you need to take min across all 3 configs. Otherwise, if config.fetchMaxBytes is lower than fetchRequest.maxBytes *and* fetchRequest.minBytes, we can still end up in an odd situation.

Reply:
fixed